File: src/main/java/com/goudourasv/data/institutions/InstitutionsMapper.java
package com.goudourasv.data.institutions;

import com.goudourasv.domain.institutions.Institution;

import java.util.ArrayList;
import java.util.List;

public class InstitutionsMapper {
    public static Institution toInstitution(InstitutionEntity institutionEntity){
        Institution institution = new Institution(institutionEntity.getId(), institutionEntity.getName(), institutionEntity.getCountry(), institutionEntity.getCity());
        return institution;
    }

    public static List<Institution> toInstitutions(List<InstitutionEntity> institutionEntities){
        List<Institution> institutions = new ArrayList<>();

        for (InstitutionEntity institutionEntity : institutionEntities) {
            Institution institution = toInstitution(institutionEntity);
            institutions.add(institution);
        }
        return  institutions;
    }
}

File: src/main/java/com/goudourasv/data/institutions/InstitutionsMapper.kt
package com.goudourasv.data.institutions

import com.goudourasv.data.instructors.InstructorEntity
import com.goudourasv.domain.institutions.Institution
import com.goudourasv.domain.institutions.InstitutionData

fun InstitutionEntity.toInstitution(): Institution {
    return Institution(
        city = this.city,
        country = this.country,
        id = this.id!!,
        name = this.name
    )
}

fun List<InstitutionEntity>.toInstitutions(): List<Institution> {
    return this.map { institutionEntity -> institutionEntity.toInstitution() }
}

fun InstitutionData.toInstitutionEntity(): InstitutionEntity{
    return InstitutionEntity(
        id = this.id,
        name = this.name,
    )
}

fun InstitutionEntity.toInstitutionData(): InstitutionData {
    return InstitutionData(
        id = this.id!!,
        name = this.name,
    )
}
